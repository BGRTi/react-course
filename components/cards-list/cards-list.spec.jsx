import { shallow } from 'enzyme';
import React from 'react';
import CardsList from './cards-list';

const filmsMock = {
  films: [
    {
      id: 1,
      name: 'first',
      year: '1991',
      duration: '151',
      description: 'Some text about film',
      genres: ['drama', 'ne drama'],
    },
  ],
};

function setup() {
  const wrapper = shallow(<CardsList films={filmsMock.films} />);
  return { wrapper };
}

describe('CardsList Test Suite', () => {
  it('render a CardsList', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });  

  it('Should have an image', () => {
    const { wrapper } = setup();

    expect(wrapper.find('.movies-container').exists()).toBe(true);
  });
});
