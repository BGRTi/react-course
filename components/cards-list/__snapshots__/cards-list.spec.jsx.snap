// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardsList Test Suite render a CardsList 1`] = `ShallowWrapper {}`;
